perceval/backends/core/discourse.py

```diff
@@ -58,20 +61,25 @@ class Discourse(Backend):
     :param sleep_time: time (in seconds) to sleep in case
         of connection problems
     """
-    version = '0.11.0'
+    version = '0.12.0'
 
     CATEGORIES = [CATEGORY_TOPIC]
     EXTRA_SEARCH_FIELDS = {
         'category_id': ['category_id']
     }
 
-    def __init__(self, url, api_token=None, tag=None, archive=None,
+    def __init__(self, url, api_username=None, api_token=None, tag=None, archive=None,
                  max_retries=MAX_RETRIES, sleep_time=DEFAULT_SLEEP_TIME):
         origin = url
 
         super().__init__(origin, tag=tag, archive=archive)
         self.url = url
+        self.api_username = api_username
         self.api_token = api_token
+
+        if (self.api_username and not self.api_token) or (self.api_token and not self.api_username):
+            raise BackendError(cause="Api token and username must be defined together")
```

Comment:
Can you add this also to the client? It would be useful to have an exception in case someone wants to use the client only.

Reply:
should this change be applied also to GitLab (https://github.com/chaoss/grimoirelab-perceval/blob/master/perceval/backends/core/gitlab.py#L106)?

Reply:
Yes!